gobblin-cluster/src/main/java/org/apache/gobblin/cluster/HelixUtils.java

```diff
@@ -251,4 +256,41 @@ private static void deleteStoppedHelixJob(HelixManager helixManager, String work
     new TaskDriver(helixManager).deleteAndWaitForCompletion(workFlowName, 10000L);
     log.info("Workflow deleted.");
   }
+
+  /**
+   * Returns the Helix Workflow Ids given {@link Iterable} of Gobblin job names. The method returns a
+   * {@link Map} from Gobblin job name to the corresponding Helix Workflow Id. This method iterates
+   * each Helix workflow, and obtains the jobs of each workflow from its jobDag.
+   * @param jobNames a list of Gobblin job names.
+   * @return a map from jobNames to their Helix Workflow Ids.
+   */
+  public static Map<String, String> getWorkflowIdsFromJobNames(HelixManager helixManager, Iterable<String> jobNames) {
+    Map<String, String> jobNameToWorkflowId = new HashMap<>();
+    TaskDriver taskDriver = new TaskDriver(helixManager);
+    Map<String, WorkflowConfig> workflowConfigMap = taskDriver.getWorkflows();
+    for (String workflow: workflowConfigMap.keySet()) {
+      WorkflowConfig workflowConfig = taskDriver.getWorkflowConfig(workflow);
+      Set<String> helixJobs = workflowConfig.getJobDag().getAllNodes();
+
+      for (String helixJob: helixJobs) {
+        Iterator<TaskConfig> taskConfigIterator = taskDriver.getJobConfig(helixJob).getTaskConfigMap().values().iterator();
+        if (taskConfigIterator.hasNext()) {
+          TaskConfig taskConfig = taskConfigIterator.next();
+          for (String jobName: jobNames) {
```

Comment:
inner loop can be converted into contains check

Reply:
workflowJobName = taskConfig.getConfigMap().get(ConfigurationKeys.JOB_NAME_KEY));
if (jobNames.contains(workflowJobName)) { ..